app/api/dao/user.py

```diff
@@ -418,3 +422,91 @@ def get_user_statistics(user_id):
+        as_mentor['received']['pending'] = [relation.response for relation in mentor_received_relations if relation.state == MentorshipRelationState.PENDING]
+
+        as_mentee['sent']['accepted'] = [relation.response for relation in mentee_sent_relations if relation.state == MentorshipRelationState.ACCEPTED]
+        as_mentee['sent']['rejected'] = [relation.response for relation in mentee_sent_relations if relation.state == MentorshipRelationState.REJECTED]
+        as_mentee['sent']['completed'] = [relation.response for relation in mentee_sent_relations if relation.state == MentorshipRelationState.COMPLETED]
+        as_mentee['sent']['cancelled'] = [relation.response for relation in mentee_sent_relations if relation.state == MentorshipRelationState.CANCELLED]
+        as_mentee['sent']['pending'] = [relation.response for relation in mentee_sent_relations if relation.state == MentorshipRelationState.PENDING]
+
+        as_mentor['sent']['accepted'] = [relation.response for relation in mentor_sent_relations if relation.state == MentorshipRelationState.ACCEPTED]
+        as_mentor['sent']['rejected'] = [relation.response for relation in mentor_sent_relations if relation.state == MentorshipRelationState.REJECTED]
+        as_mentor['sent']['completed'] = [relation.response for relation in mentor_sent_relations if relation.state == MentorshipRelationState.COMPLETED]
+        as_mentor['sent']['cancelled'] = [relation.response for relation in mentor_sent_relations if relation.state == MentorshipRelationState.CANCELLED]
+        as_mentor['sent']['pending'] = [relation.response for relation in mentor_sent_relations if relation.state == MentorshipRelationState.PENDING]
+
+        response['as_mentor'] = as_mentor
+        response['as_mentee'] = as_mentee
+
+        current_relation = MentorshipRelationDAO.list_current_mentorship_relation(user_id=user_id)
+        response['tasks_todo'] = marshal([task for task in current_relation.tasks_list.tasks if not task['is_done']], list_tasks_response_body)
+        response['tasks_done'] = marshal([task for task in current_relation.tasks_list.tasks if task['is_done']], list_tasks_response_body)
+
+        return response
+
+
+class DashboardRelationResponseModel:
+    """temp class used for storing mentorship_request_response_body_for_user_dashboard_body values"""
+    def __init__(self, relation: MentorshipRelationModel):
+        self.state = relation.state
+        self.mentor_id = relation.mentor_id
+        self.mentee_id = relation.mentee_id
+        self.action_user_id = relation.action_user_id
+        self.response = {
+            'id': relation.id,
+            'action_user_id': relation.action_user_id,
+            'mentor_id': relation.mentor.id,
+            'mentee_id': relation.mentee.id,
+            'mentor_name': relation.mentor.name,
+            'mentee_name': relation.mentee.name,
+            'mentor_profile_picture_URL': relation.mentor.photo_url,
```

Comment:
I would keep the convention of small url. I would put mentor and mentee as sub dictionary, so you could have helper method for parsing user. What do you think?

Reply:
fixed the small url thing

Reply:
could you please explain what you mean by putting mentor and mentee in sub dictionary?